.github/workflows/release.yml

```diff
@@ -1,113 +1,52 @@
-        echo "version=$(./.github/ci/next_version.sh)" >> "$GITHUB_OUTPUT"
-  
-  draft_release:
-    runs-on: ubuntu-latest
-    needs: [version]
-    steps:
-      - uses: actions/checkout@v3
-
-      # Explicitly pass in the sha, even though it's very likely to be HEAD.
-      - run: |
-          hub release create \
-            --prerelease \
-            --draft \
-            --message "$(./.github/ci/generate_release_message.sh ${{ needs.version.outputs.version }})" \
-            -t ${{ github.sha }} \
-            "${{ needs.version.outputs.version }}"
-
-        # Propagate token into environment
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-  # Even though the release job has an on: release trigger, actions taken by GITHUB_TOKEN
-  # don't trigger this, so we call it here
-  # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
   release:
-    needs: [draft_release, version]
-    uses: ./.github/workflows/release_artifacts.yml
-    with:
-      version: ${{needs.version.outputs.version}}
+    runs-on: ubuntu-latest
     secrets:
-      TIGERBEETLE_NODE_PUBLISH_KEY: ${{secrets.TIGERBEETLE_NODE_PUBLISH_KEY}}
-      NUGET_KEY: ${{secrets.NUGET_KEY}}
-      MAVEN_GPG_SECRET_KEY: ${{ secrets.MAVEN_GPG_SECRET_KEY }}
-      MAVEN_CENTRAL_USERNAME: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
-      MAVEN_CENTRAL_TOKEN: ${{ secrets.MAVEN_CENTRAL_TOKEN }}
-      MAVEN_GPG_SECRET_KEY_PASSWORD: ${{ secrets.MAVEN_GPG_SECRET_KEY_PASSWORD }}
-      TIGERBEETLE_GO_DEPLOY_KEY: ${{ secrets.TIGERBEETLE_GO_DEPLOY_KEY }}
 
-  finalize_release:
-    runs-on: ubuntu-latest
-    needs: [release, version]
     steps:
       - uses: actions/checkout@v3
 
-      - run: |
-          hub release edit \
-            --draft=false \
-            -m "" \
-            "${{ needs.version.outputs.version }}"
-
-        # Propagate token into environment
-        env:
-          EDITOR: vi
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-  docs:
-    needs: [release, version]
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - run: ./.github/ci/publish_docs.sh
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: '3.1.x'
+
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_CENTRAL_TOKEN
+          gpg-private-key: ${{ secrets.MAVEN_GPG_SECRET_KEY }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+      # No special setup for Go.
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - run: ./scripts/install_zig.sh
+      - run: ./zig/zig build dist -- --build --publish --version 0.9.${{ github.run_number }} --sha ${{ github.sha }}
```

Comment:
Both `--build` and `--publish` at the same time for now. That's the place where we would be able to insert `--and-thoroughly-test` in the future. This is the single place where we might want to use `upload-artifact` action.

Reply:
> That's the place where we would be able to insert --and-thoroughly-test in the future.

(if you're speaking about `--and-thoroughly-test` testing still-bounded-tests-that-take-a-while as opposed to things like fuzzing, VOPR, than the below doesn't apply!)

Jobs are limited to ~6 hours on GitHub, so we would need to split it up (also, we don't want to incur the waste of an idle runner while the tests are really being run on different hardware!)